src/main/java/com/ays/assignment/model/dto/request/UserAssignmentListRequest.java

```diff
@@ -0,0 +1,79 @@
+package com.ays.assignment.model.dto.request;
+
+import com.ays.assignment.model.enums.AssignmentStatus;
+import com.ays.common.model.dto.request.AysFilteringRequest;
+import com.ays.common.model.dto.request.AysPagingRequest;
+import com.ays.common.util.validation.EnumValidation;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import jakarta.validation.constraints.AssertTrue;
+import lombok.Builder;
+import lombok.Data;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.EnumSet;
+import java.util.Set;
+
+/**
+ * Represents a request object for fetching a list of user assignment with pagination,sorting and filtering options
+ * This class extends the {@link AysPagingRequest} class and adds additional validation rules for sorting.
+ */
+@Data
+@Builder
+public class UserAssignmentListRequest extends AysPagingRequest implements AysFilteringRequest {
+
+    private AssignmentStatus status;
+
+    /**
+     * Checks if the assignment status is valid.
+     *
+     * @return true if the assignment status is valid or null, false otherwise.
+     */
+    @AssertTrue(message = "IS ASSIGNMENT STATUS NOT ACCEPTED")
+    private boolean isStatusAccepted() {
+
+        if (this.status == null) {
+            return true;
+        }
+
+        EnumSet<AssignmentStatus> acceptedAssignmentStatuses = EnumSet.of(AssignmentStatus.AVAILABLE,
+                AssignmentStatus.RESERVED,
+                AssignmentStatus.ASSIGNED,
+                AssignmentStatus.IN_PROGRESS,
+                AssignmentStatus.DONE
+        );
+        return EnumValidation.anyOf(this.status, acceptedAssignmentStatuses);
+    }
```

Comment:
Listleme yaparken sadece buradaki statü bilgileriyle mi listeleme yapabilecek yoksa tüm bilgilerle listleme yapabilecek mi?

Reply:
O durum için `toSpecification` kullanmıştım.